### PR TITLE
Merge FrontendTest to Master

### PR DESCRIPTION
**Pull request by coreylevine**
- Added notification functionalities.
- Added TA functionalities.

### CODE REVIEW

**Comment by rekandiah** [comment]
error

There's an error with the login button. It's only able to click it at a certain location. I think its due the interference of the waves in the back? I tried it manually as well and I get the same issue.

@j-suguma @jameelbrochu

File: test/components/app_test.js [hunk 1]
```diff
@@ -2,30 +2,56 @@ import {renderComponent, expect} from '../test_helper';
...
-    it('number of days', () => {
-    expect(noClassThisDay.length).equals(11);
+});
+
+describe('Weekdays', () => {
+  let startDate = new Date(2017, 1, 9);
+  let endDate = new Date(2017, 4, 13);
+  let noClassThisDay = new Array();
+  FireBaseTools.fillNoClassThisDay(noClassThisDay);
+  let val = () => { return {
```
Comment (mukulikadey) [inline test/components/app_test.js]: @coreylevine What does this val do? Why don't you just put it all in snap?

**Comment by jameelbrochu** [comment]
@rekandiah will look into. We will mostly like have to remove the wave because the movement of it is what is causing certains parts of the login and register buttons to become disable when it passed over it.

File: test/components/app_test.js (hunk 1 above)
Comment (coreylevine) [inline test/components/app_test.js]: We need to get snap.val() for the function checkWeekday and so in order to do this we make val a lambda function and make snap is a variable containing a lambda function.

Comment (mukulikadey) [inline test/components/app_test.js]: Ahh okay makes sense !

**Comment by svelupillai** [comment]
@jameelbrochu I fixed the issue, there was also a problem with the notifications in the nav bar in the app.jsx file it made the site crash. I am going to push these changes

**Pull request merged**